packages/core/src/app/coupon/useMultiCoupon.ts

```diff
@@ -1,44 +1,97 @@
 import { useState } from 'react';
 
-import { useCheckout } from '@bigcommerce/checkout/contexts';
+import { useCheckout, useLocale } from '@bigcommerce/checkout/contexts';
 
 import { EMPTY_ARRAY } from '../common/utility';
 
+import { type AppliedGiftCertificateInfo } from './components/AppliedGiftCertificates';
+
+export interface DiscountItem {
+    name: string;
+    amount: number;
+}
+
+interface UIDetails {
+    subtotal: number;
+    discounts: number;
+    discountItems: DiscountItem[];
+    shipping: number;
+    shippingBeforeDiscount: number;
+}
+
 interface UseMultiCouponValues {
     appliedCoupons: Array<{ code: string }>;
-    appliedGiftCertificates: Array<{ code: string }>;
-    applyCouponOrGiftCertificate: (code: string) => Promise<void>;
+    appliedGiftCertificates: AppliedGiftCertificateInfo[];
     couponError: string | null;
-    isCouponCodeCollapsed: boolean;
+    isCouponFormCollapsed: boolean;
```

Comment:
Is there a reason we are storing UI states in a custom hook?

Reply:
It's the default display value for the coupon form, read from `checkoutSettings`.
```
isCouponFormCollapsed: checkoutSettings.isCouponCodeCollapsed,
```